### PR TITLE
Fix validation

### PR DESCRIPTION
**Pull request by xepozz**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: tests/Unit/Helpers/DefinitionValidatorTest.php [hunk 1]
```diff
@@ -98,23 +119,23 @@ public function testInvalidMethodCalls(array $methodCalls, string $message): voi
...
-            [' var$', true, '~^Invalid definition: key " var\$" is not allowed\. Did you mean "var\(\)" or "\$var"\?$~'],
-            ['100$', true, '~^Invalid definition: key "100\$" is not allowed\.$~'],
-            ['test-тест', true, '~^Invalid definition: key "test-тест" is not allowed\.$~'],
+            ['dev', true, 'Invalid definition: key "dev" is not allowed. Did you mean "dev()" or "$dev"?'],
+            ['setId', [42], 'Invalid definition: key "setId" is not allowed. Did you mean "setId()" or "$setId"?'],
+            ['()test', [42], 'Invalid definition: key "()test" is not allowed. Did you mean "test()" or "$test"?'],
+            ['var$', true, 'Invalid definition: key "var$" is not allowed. Did you mean "var()" or "$var"?'],
+            [' var$', true, 'Invalid definition: key " var$" is not allowed. Did you mean "var()" or "$var"?'],
+            ['100$', true, 'Invalid definition: key "100$" is not allowed.'],
+            ['test-тест', true, 'Invalid definition: key "test-тест" is not allowed.'],
```
Comment (vjik) [inline tests/Unit/Helpers/DefinitionValidatorTest.php:128]: Why this change?

After this change added mutants.

File: tests/Unit/Helpers/DefinitionValidatorTest.php [hunk 2]
```diff
@@ -32,8 +32,8 @@ public function testIntegerKeyOfArray(): void
     public function dataInvalidClass(): array
     {
         return [
-            [42, 'Invalid definition: invalid class name. Expected string, got int'],
-            ['', 'Invalid definition: empty class name.'],
+            [42, 'Invalid definition: class name must be a non-empty string, got "int".'],
+            ['', 'Invalid definition: class name must be a non-empty string, got "string".'],
```
Comment (vjik) [inline tests/Unit/Helpers/DefinitionValidatorTest.php:36]: May be for empty string make unique message? Current message is confusing.

**Review by vjik — CHANGES_REQUESTED** [review]
Overall good!

This package is mutants free. Can you kill new mutants?

File: tests/Unit/Helpers/DefinitionValidatorTest.php (hunk 2 above)
Comment (xepozz) [reply tests/Unit/Helpers/DefinitionValidatorTest.php:36]: I've fixed it [there](https://github.com/yiisoft/definitions/pull/57)

File: tests/Unit/Helpers/DefinitionValidatorTest.php (hunk 1 above)
Comment (xepozz) [reply tests/Unit/Helpers/DefinitionValidatorTest.php:128]: There is no need in regular expressions

Comment (vjik) [reply tests/Unit/Helpers/DefinitionValidatorTest.php:128]: Need check message full. Without regex pass message with any suffix. 

See mutants.

File: tests/Unit/Helpers/DefinitionValidatorTest.php (hunk 2 above)
Comment (vjik) [reply tests/Unit/Helpers/DefinitionValidatorTest.php:36]: Ok
